main.c

```diff
@@ -1,13 +1,40 @@
 #include <stdio.h>
+#include <stdlib.h>
+
+struct Node {
+    int data;
+    struct Node* next;
+};
 
 int main() {
-    // TODO: build a linked-list version of this array
-    int numbers[] = {1, 2, 3, 4, 5};
 
-    for (int i = 0; i < 5; i++) {
-        printf("%d\n", numbers[i]);
+    struct Node* head = (struct Node*)malloc(sizeof(struct Node));
+    struct Node* second = (struct Node*)malloc(sizeof(struct Node));
+    struct Node* third = (struct Node*)malloc(sizeof(struct Node));
+    struct Node* fourth = (struct Node*)malloc(sizeof(struct Node));
+    struct Node* fifth = (struct Node*)malloc(sizeof(struct Node));
+
+    head->data = 1;
+    head->next = second;
+
+    second->data = 2;
+    second->next = third;
+
+    third->data = 3;
+    third->next = fourth;
+
+    fourth->data = 4;
+    fourth->next = fifth;
+
+    fifth->data = 5;
+    fifth->next = NULL;
+
+    struct Node* current = head;
+    while (current != NULL) {
+        printf("%d\n", current->data);
```

Comment:
It's good practice to free the dynamically allocated memory after you're done using it to avoid memory leaks. Add a section at the end of your `main` function to free the nodes.